ProjFS.Mac/PrjFSKextTests/HandleOperationTests.mm

```diff
@@ -481,10 +485,12 @@ - (void) testDeleteFileNonRenamed
         _,
         _,
         nullptr));
+    CleanupPendingRenames();
```

Comment:
The (new with this PR) `s_osSupportsRenameDetection` variable could end up being in an inconsistent state for these tests otherwise. I was on the fence whether to set `s_osSupportsRenameDetection` when setting `version_major` instead, but relying on uninitialised pending rename array state in the tests isn't great anyway. We specifically *don't* want to call `InitPendingRenames();`/`CleanupPendingRenames();` from `setUp`/`tearDown` because their version-specific so they need to be called inside the block passed to `TestForAllSupportedDarwinVersions()`, not before/after the whole test method.

It's not a super nice solution, I'll grant you that. In an ideal world we'd probably do some fancy stuff to run the whole test suite for each supported `major_version` instead of doing that on a test-by-test basis, but I don't have an immediate suggestion on how to do that with XCTest.